comm/receiver.go

```diff
@@ -0,0 +1,8 @@
+package comm
+
+// Receiver - Contains methods for receiving data over the network
+type Receiver interface {
+	NewSocket(addr string) error
+	Receive() ([]byte, error)
+	Close() error
+}
```

Comment:
Top, Wellington! Era bem esse conceito mesmo que eu precisava. Porém, nossa interface só precisa ter os métodos Receive e Close nela. O New deve ser particulare de cada tipo. Abaixo vai uma explicação mais detalhada:

Pensa que nosso sistema vai ter mais ou menos a seguinte estrutura:
`type System struct {
          receiver Receiver
          decoder Decoder
          evaluator Evaluator
          ...
}`

Sendo assim, quando a gente for instanciar um System, precisamos ter um Receiver criado ele anteriormente, para passarmos o ponteiro do mesmo (ou até mesmo podemos passar o método particular, por exemplo: NewUDPReceiver("localhost:8080")). É justamente para isso que funciona a interface, assim podemos ter mais de um tipo de Receiver e escolhermos ele dinamicamente!

Reply:
Entendi. Mas para todos os senders e receivers não vai haver um construtor? Sendo ele udp, zmq, comunicação serial, radio, etc? então, se todos eles precisam de um construtor, eles não respeitariam o contrato da interface, podendo assim ser a mesma interface para todos?

mas isso é só por dúvida msm, jaja eu arrumo

Reply:
Sim, respeita sim. Assim também tá funcional! É uma questão de seguirmos nosso padrão de legibilidade.